scripts/generate

```diff
@@ -480,45 +503,28 @@ def('toc',
     [$.String, $.String],
     pipe([matchAll(/<(h[1-6]) id="([^"]*)">(.*)<[/]\1>/g),
           map(prop('groups')),
-          map(justs),
-          reduce(({level, tagName, html}) => ([hN, id, _innerHtml]) => {
-            const level$ = Number(hN[1]);
-            const level$$ = level$ > level ? hN === tagName ? level : level + 1 : level$;
-
-            const pattern = '<a [^>]*>([^<]*)</a>';
-            const innerHtml =
-            pipe([replace(regex('', pattern), '<b>$1</b>'),
-                  replace(regex('g', pattern), '$1')],
-                 _innerHtml);
-
-            const html$ =
-            html + '\n' +
-            (level$$ > level ?
-               pad(2 * level$$ - 2) + '<ul' + (level === 1 ? ' id="toc"' : '') + '>\n' +
-               pad(2 * level$$ - 1) + '<li>\n' +
-               pad(2 * level$$ - 0) :
-             level$$ < level ?
-               pad(2 * level$$ + 1) + '</li>\n' +
-               pad(2 * level$$ - 0) + '</ul>\n' +
-               pad(2 * level$$ - 1) + '</li>\n' +
-               pad(2 * level$$ - 1) + '<li>\n' +
-               pad(2 * level$$ - 0) :
-             // else
-               pad(2 * level$$ - 1) + '</li>\n' +
-               pad(2 * level$$ - 1) + '<li>\n' +
-               pad(2 * level$$ - 0)) +
-            el('a', {href: '#' + id}, innerHtml);
-
-            return {level: level$$, tagName: hN, html: html$};
-          }, {level: 1, tagName: 'h1', html: ''}),
-          over(lensProp('level'),
-               compose(j,
-                       unfoldr(level => level > 1 ?
-                                 Just(['\n' + pad(2 * level - 1) + '</li>' +
-                                       '\n' + pad(2 * level - 2) + '</ul>',
-                                      level - 1]) :
-                                 Nothing))),
-          lift2(concat, prop('html'), prop('level'))]));
+          map(map(fromJust)),
```

Comment:
Semantically, `map (fromJust)` is more appropriate than `justs` because it asserts that none of the groups is `Nothing`. This in turn ensures that the `[tagName, id, html]` destructuring will succeed.